cfme/physical/physical_storage.py

```diff
@@ -0,0 +1,231 @@
+        wait_for(_is_state_changed, fail_func=view.browser.refresh, num_sec=timeout, delay=delay)
+
+    @property
+    def exists(self):
+        """Checks if the physical_Storage exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        view = navigate_to(self.parent, "All")
+        try:
+            view.entities.get_entity(name=self.name, surf_pages=True)
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    @cached_property
+    def db_id(self):
+        return self.appliance.physical_storage_id(self.name)
+
+    def wait_to_appear(self):
+        """Waits for the storage to appear in the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the storage to appear...")
+        wait_for(
+            lambda: self.exists,
+            message="Wait for the storage to appear",
+            num_sec=1000,
+            fail_func=view.browser.refresh
+        )
+
+    def wait_for_delete(self):
+        """Waits for the storage to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the storage to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the storage to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class PhysicalStorageCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.PhysicalStorage`."""
+
+    ENTITY = PhysicalStorage
+
+    def select_entity_rows(self, *physical_storages):
+        """ Select all physical storage objects """
+        physical_storages = list(physical_storages)
+        checked_physical_storages = list()
+        view = navigate_to(self, 'All')
+
+        for physical_storage in physical_storages:
+            view.entities.get_entity(name=physical_storage.name, surf_pages=True).check()
+            checked_physical_storages.append(physical_storage)
+        return view
+
+    def all(self):
+        """returning all physical_storages objects"""
+        physical_storage_table = self.appliance.db.client['physical_storages']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        physical_storage_query = (
+            self.appliance.db.client.session
+                .query(physical_storage_table.name, ems_table.name)
+                .join(ems_table, physical_storage_table.id == ems_table.id))
+        provider = self.filters.get('provider')
+        if provider:
+            physical_storage_query = physical_storage_query.filter(ems_table.name == provider.name)
+        physical_storages = []
+        for name, ems_name in physical_storage_query.all():
+            physical_storages.append(self.instantiate(name=name,
+                                    provider=provider or get_crud_by_name(ems_name)))
+        return physical_storages
+
+    def find_by(self, ph_name, provider=None):
```

Comment:
Question 
Why you need this method? You have all method doing the same thing for you. You can filter out in a test as well.